docs/guide/request_lifecycle.md

```diff
@@ -0,0 +1,155 @@
+
+As Ibexa DXP is a Symfony application, everything starts like in Symfony.
+- https://symfony.com/doc/current/introduction/http_fundamentals.html#requests-and-responses-in-http
+- https://symfony.com/doc/current/introduction/http_fundamentals.html#the-symfony-application-flow
+
+If the HTTP request is to be treated by Ibexa DXP, it goes to the [Symfony Front Controller](https://symfony.com/doc/current/configuration/front_controllers_and_kernel.html#the-front-controller) public/index.php
+
+The Front Controller transform the HTTP request into a PHP object `Request` and pass it to Symfony's Kernel to obtain a `Response` object that will be sent back as an HTTP response.
+
+The schemas start with a Request object entering Symfony and Ibexa DXP. No REST or GraphQL HTTP Request here, only regular request from a browser.
+
+
+
+## From Request to Response: Flowcharts
+
+### Concept Flowchart
+
+![](img/request_lifecycle_concept.png)
+
+### Kernel Events Flowchart
+
+![](img/request_lifecycle_events.png)
+
+
+
+## Kernel's Request Event
+
+When the Request enter the Symfony's Kernel (and underneath the [HttpKernel](https://symfony.com/doc/current/components/http_kernel.html)), a `kernel.request` event (a.k.a `KernelEvents::REQUEST`) is dispatched.
+Several listeners are called by decreasing priority (see `bin/console debug:event-dispatcher kernel.request;`).
+
+### SiteAccess Matching
+
+After the FragmentListener (priority 48), this is the turn of the `ezpublish.siteaccess_match_listener` service (priority 45).
+This service can be
+- purely the SiteAccessMatchListener or
+- its UserContextSiteAccessMatchSubscriber decoration when HTTP Cache is used.
+  This service will
+- find the current siteaccess using the SiteAccess\Router regarding the [SiteAccess Matching configurations](https://doc.ibexa.co/en/latest/guide/multisite/siteaccess_matching/),
+- add the current siteaccess to the Request object's attribute `siteaccess`,
+- then dispatch the `ezpublish.siteaccess` event (a.k.a MVCEvents::SITEACCESS) (see `bin/console debug:event-dispatcher ezpublish.siteaccess;`).
+
+The SiteAccessListener subscribes to this `ezpublish.siteaccess` event with top priority. The SiteAccessListener add the attribute `semanticPathinfo`, the path without siteaccess indications ([URIElement](https://doc.ibexa.co/en/latest/guide/multisite/siteaccess_matching/#urielement), [URIText](https://doc.ibexa.co/en/latest/guide/multisite/siteaccess_matching/#uritext) or [Map\URI](https://doc.ibexa.co/en/latest/guide/multisite/siteaccess_matching/#mapuri) implementing the URILexer interface), to the Request.
+
+
+### Routing
+
+Lately, also listening the `kernel.request` event, the Symfony\Component\HttpKernel\EventListener\RouterListener (priority 32) calls the eZ\Publish\Core\MVC\Symfony\Routing\ChainRouter::matchRequest and add its returned parameters to the Request.
+
+ChainRouter:
```

Comment:
these deserve a heading, I think
```suggestion
#### ChainRouter

```

Reply:
Done. But only for the ChainRouter. Maybe Default, UrlWildcardRouter and UrlAliasRouter deserve a heading too.

Reply:
Done